spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -398,13 +399,13 @@ static List<String> findOrphanFiles(
     spark.sparkContext().register(conflicts);
 
     Column joinCond = actualFileIdentDS.col("path").equalTo(validFileIdentDS.col("path"));
-
-    List<String> orphanFiles =
+    PatriciaTrie<String> files = new PatriciaTrie<>();
```

Comment:
We don't really leverage the efficient search provided by tries for orphan files; rather I'm looking into the memory saving aspects of this structure when prefixes are repeated many times.